// bootstrap-vue-next/dist/index-NYJ0QNUy.mjs
import { _ as _sfc_main, a as _sfc_main$1 } from "./BAccordion.vue_vue_type_script_setup_true_lang-Bm1RghW9.mjs";
import { _ as _sfc_main$2 } from "./BAlert.vue_vue_type_script_setup_true_lang-BEAKw4MY.mjs";
import { _ as _sfc_main$3 } from "./BApp.vue_vue_type_script_setup_true_lang-CRI7YgFM.mjs";
import { _ as _sfc_main$1b } from "./BOrchestrator.vue_vue_type_script_setup_true_lang-Dz_rJ4Bq.mjs";
import { _ as _sfc_main$4, a as _sfc_main$5 } from "./BAvatarGroup.vue_vue_type_script_setup_true_lang-DGkjFwcd.mjs";
import { _ as _sfc_main$6 } from "./BBadge.vue_vue_type_script_setup_true_lang-D9DEOOtU.mjs";
import { _ as _sfc_main$7, a as _sfc_main$8 } from "./BBreadcrumb.vue_vue_type_script_setup_true_lang-D6cRQgRH.mjs";
import { _ as _sfc_main$9 } from "./BButton.vue_vue_type_script_setup_true_lang-D5Eg7-Zv.mjs";
import { _ as _sfc_main$a, a as _sfc_main$b } from "./BButtonToolbar.vue_vue_type_script_setup_true_lang-CxCcFupr.mjs";
import { _ as _sfc_main$n } from "./BCloseButton.vue_vue_type_script_setup_true_lang-byZmQ_ot.mjs";
import { _ as _sfc_main$c, a as _sfc_main$d, b as _sfc_main$e, c as _sfc_main$g, d as _sfc_main$h, e as _sfc_main$i, f as _sfc_main$k } from "./BCard.vue_vue_type_script_setup_true_lang-BjoxuViz.mjs";
import { _ as _sfc_main$f, a as _sfc_main$j } from "./BCardText.vue_vue_type_script_setup_true_lang-SRGbviaO.mjs";
import { _ as _sfc_main$l, a as _sfc_main$m } from "./BCarouselSlide.vue_vue_type_script_setup_true_lang-Cx8lSNLo.mjs";
import { _ as _sfc_main$p } from "./BCollapse.vue_vue_type_script_setup_true_lang-B8TqF-kB.mjs";
import { _ as _sfc_main$o } from "./BCol.vue_vue_type_script_setup_true_lang-D0xTJQ30.mjs";
import { _ as _sfc_main$q } from "./BContainer.vue_vue_type_script_setup_true_lang-Bi1hGl0B.mjs";
import { _ as _sfc_main$1n } from "./BRow.vue_vue_type_script_setup_true_lang-BDkBauhk.mjs";
import { _ as _sfc_main$r } from "./BDropdown.vue_vue_type_script_setup_true_lang-D1wLUc9_.mjs";
import { _ as _sfc_main$s, a as _sfc_main$t, b as _sfc_main$u, c as _sfc_main$v, d as _sfc_main$w, e as _sfc_main$x, f as _sfc_main$y } from "./BDropdownText.vue_vue_type_script_setup_true_lang-Cm6QYE9x.mjs";
import { _ as _sfc_main$z } from "./BForm.vue_vue_type_script_setup_true_lang-u44ElAm9.mjs";
import { _ as _sfc_main$C, a as _sfc_main$E } from "./BFormFloatingLabel.vue_vue_type_script_setup_true_lang-Bnwbg5nF.mjs";
import { _ as _sfc_main$H, a as _sfc_main$L, b as _sfc_main$S, c as _sfc_main$U } from "./BFormValidFeedback.vue_vue_type_script_setup_true_lang-D5WYd1j7.mjs";
import { _ as _sfc_main$A, a as _sfc_main$B } from "./BFormCheckboxGroup.vue_vue_type_script_setup_true_lang-CZ2qJTp0.mjs";
import { _ as _sfc_main$D } from "./BFormFile.vue_vue_type_script_setup_true_lang-BF3fBjJ3.mjs";
import { _ as _sfc_main$F } from "./BFormGroup.vue_vue_type_script_setup_true_lang-BmZN8qcE.mjs";
import { _ as _sfc_main$G } from "./BFormInput.vue_vue_type_script_setup_true_lang-D3z3jxbI.mjs";
import { _ as _sfc_main$I, a as _sfc_main$J } from "./BFormRadioGroup.vue_vue_type_script_setup_true_lang-LQBMttYK.mjs";
import { _ as _sfc_main$K } from "./BFormRating.vue_vue_type_script_setup_true_lang-DxkXJngG.mjs";
import { _ as _sfc_main$M, a as _sfc_main$O } from "./BFormSelect.vue_vue_type_script_setup_true_lang-CkLgnxVA.mjs";
import { _ as _sfc_main$N } from "./useFormSelect-DvaFh_iu.mjs";
import { _ as _sfc_main$P } from "./BFormSpinbutton.vue_vue_type_script_setup_true_lang-BZZ4vW74.mjs";
import { a as _sfc_main$Q, _ as _sfc_main$R } from "./BFormTags.vue_vue_type_script_setup_true_lang-C9rQBsnw.mjs";
import { _ as _sfc_main$T } from "./BFormTextarea.vue_vue_type_script_setup_true_lang-DB0GbEW7.mjs";
import { _ as _sfc_main$V } from "./BImg.vue_vue_type_script_setup_true_lang-DFoA1dqz.mjs";
import { _ as _sfc_main$W, a as _sfc_main$X } from "./BInputGroupText.vue_vue_type_script_setup_true_lang-BBN5F0A4.mjs";
import { _ as _sfc_main$Y } from "./BLink.vue_vue_type_script_setup_true_lang-Ct47WhzE.mjs";
import { _ as _sfc_main$Z, a as _sfc_main$_ } from "./BListGroupItem.vue_vue_type_script_setup_true_lang-J0Ydbiva.mjs";
import { _ as _sfc_main$$ } from "./index-47fMl4-P.mjs";
import { _ as _sfc_main$10 } from "./BModalOrchestrator.vue_vue_type_script_setup_true_lang-DjUWzuI9.mjs";
import { _ as _sfc_main$11, a as _sfc_main$12, b as _sfc_main$13, c as _sfc_main$14, d as _sfc_main$15 } from "./BNavText.vue_vue_type_script_setup_true_lang-fCkr2EkB.mjs";
import { _ as _sfc_main$16, a as _sfc_main$17, b as _sfc_main$18, c as _sfc_main$19 } from "./BNavbarToggle.vue_vue_type_script_setup_true_lang-BOVCTLjN.mjs";
import { _ as _sfc_main$1a } from "./BOffcanvas.vue_vue_type_script_setup_true_lang-DMDfcqoh.mjs";
import { _ as _sfc_main$1c } from "./BOverlay.vue_vue_type_script_setup_true_lang-B5_buAVa.mjs";
import { _ as _sfc_main$1d } from "./BPagination.vue_vue_type_script_setup_true_lang-C0Xo-KSI.mjs";
import { _ as _sfc_main$1e, a as _sfc_main$1f, b as _sfc_main$1g, c as _sfc_main$1h, d as _sfc_main$1i } from "./BPlaceholderWrapper.vue_vue_type_script_setup_true_lang-BzabzEJo.mjs";
import { _ as _sfc_main$1j } from "./floatingUi-BBabBWxD.mjs";
import { _ as _sfc_main$1k } from "./BPopoverOrchestrator.vue_vue_type_script_setup_true_lang-BME1-ILp.mjs";
import { _ as _sfc_main$1l, a as _sfc_main$1m } from "./BProgress.vue_vue_type_script_setup_true_lang-CzGUiUCY.mjs";
import { _ as _sfc_main$1o } from "./BSpinner.vue_vue_type_script_setup_true_lang-H1MzHypA.mjs";
import { _ as _sfc_main$1q, a as _sfc_main$1r, b as _sfc_main$1u, c as _sfc_main$1v, d as _sfc_main$1w, e as _sfc_main$1x, f as _sfc_main$1y, g as _sfc_main$1C } from "./BTable.vue_vue_type_script_setup_true_lang-BzqnAvaM.mjs";
import { _ as _sfc_main$1s } from "./BTableSimple.vue_vue_type_script_setup_true_lang-pOlbJbSz.mjs";
import { _ as _sfc_main$1p, a as _sfc_main$1t } from "./BTabs.vue_vue_type_script_setup_true_lang-DJniJmO4.mjs";
import { _ as _sfc_main$1z } from "./index-7PG2ESMq.mjs";
import { _ as _sfc_main$1A } from "./BToastOrchestrator.vue_vue_type_script_setup_true_lang-Dg1phGTf.mjs";
import { _ as _sfc_main$1B } from "./BTooltip.vue_vue_type_script_setup_true_lang-DwLY6gwe.mjs";
const index = /* @__PURE__ */ Object.freeze(/* @__PURE__ */ Object.defineProperty({
  __proto__: null,
  BAccordion: _sfc_main,
  BAccordionItem: _sfc_main$1,
  BAlert: _sfc_main$2,
  BApp: _sfc_main$3,
  BAvatar: _sfc_main$4,
  BAvatarGroup: _sfc_main$5,
  BBadge: _sfc_main$6,
  BBreadcrumb: _sfc_main$7,
  BBreadcrumbItem: _sfc_main$8,
  BButton: _sfc_main$9,
  BButtonGroup: _sfc_main$a,
  BButtonToolbar: _sfc_main$b,
  BCard: _sfc_main$c,
  BCardBody: _sfc_main$d,
  BCardFooter: _sfc_main$e,
  BCardGroup: _sfc_main$f,
  BCardHeader: _sfc_main$g,
  BCardImg: _sfc_main$h,
  BCardSubtitle: _sfc_main$i,
  BCardText: _sfc_main$j,
  BCardTitle: _sfc_main$k,
  BCarousel: _sfc_main$l,
  BCarouselSlide: _sfc_main$m,
  BCloseButton: _sfc_main$n,
  BCol: _sfc_main$o,
  BCollapse: _sfc_main$p,
  BContainer: _sfc_main$q,
  BDropdown: _sfc_main$r,
  BDropdownDivider: _sfc_main$s,
  BDropdownForm: _sfc_main$t,
  BDropdownGroup: _sfc_main$u,
  BDropdownHeader: _sfc_main$v,
  BDropdownItem: _sfc_main$w,
  BDropdownItemButton: _sfc_main$x,
  BDropdownText: _sfc_main$y,
  BForm: _sfc_main$z,
  BFormCheckbox: _sfc_main$A,
  BFormCheckboxGroup: _sfc_main$B,
  BFormDatalist: _sfc_main$C,
  BFormFile: _sfc_main$D,
  BFormFloatingLabel: _sfc_main$E,
  BFormGroup: _sfc_main$F,
  BFormInput: _sfc_main$G,
  BFormInvalidFeedback: _sfc_main$H,
  BFormRadio: _sfc_main$I,
  BFormRadioGroup: _sfc_main$J,
  BFormRating: _sfc_main$K,
  BFormRow: _sfc_main$L,
  BFormSelect: _sfc_main$M,
  BFormSelectOption: _sfc_main$N,
  BFormSelectOptionGroup: _sfc_main$O,
  BFormSpinbutton: _sfc_main$P,
  BFormTag: _sfc_main$Q,
  BFormTags: _sfc_main$R,
  BFormText: _sfc_main$S,
  BFormTextarea: _sfc_main$T,
  BFormValidFeedback: _sfc_main$U,
  BImg: _sfc_main$V,
  BInput: _sfc_main$G,
  BInputGroup: _sfc_main$W,
  BInputGroupText: _sfc_main$X,
  BLink: _sfc_main$Y,
  BListGroup: _sfc_main$Z,
  BListGroupItem: _sfc_main$_,
  BModal: _sfc_main$$,
  BModalOrchestrator: _sfc_main$10,
  BNav: _sfc_main$11,
  BNavForm: _sfc_main$12,
  BNavItem: _sfc_main$13,
  BNavItemDropdown: _sfc_main$14,
  BNavText: _sfc_main$15,
  BNavbar: _sfc_main$16,
  BNavbarBrand: _sfc_main$17,
  BNavbarNav: _sfc_main$18,
  BNavbarToggle: _sfc_main$19,
  BOffcanvas: _sfc_main$1a,
  BOrchestrator: _sfc_main$1b,
  BOverlay: _sfc_main$1c,
  BPagination: _sfc_main$1d,
  BPlaceholder: _sfc_main$1e,
  BPlaceholderButton: _sfc_main$1f,
  BPlaceholderCard: _sfc_main$1g,
  BPlaceholderTable: _sfc_main$1h,
  BPlaceholderWrapper: _sfc_main$1i,
  BPopover: _sfc_main$1j,
  BPopoverOrchestrator: _sfc_main$1k,
  BProgress: _sfc_main$1l,
  BProgressBar: _sfc_main$1m,
  BRow: _sfc_main$1n,
  BSpinner: _sfc_main$1o,
  BTab: _sfc_main$1p,
  BTable: _sfc_main$1q,
  BTableLite: _sfc_main$1r,
  BTableSimple: _sfc_main$1s,
  BTabs: _sfc_main$1t,
  BTbody: _sfc_main$1u,
  BTd: _sfc_main$1v,
  BTfoot: _sfc_main$1w,
  BTh: _sfc_main$1x,
  BThead: _sfc_main$1y,
  BToast: _sfc_main$1z,
  BToastOrchestrator: _sfc_main$1A,
  BTooltip: _sfc_main$1B,
  BTr: _sfc_main$1C
}, Symbol.toStringTag, { value: "Module" }));
export {
  index as i
};
